### PR TITLE
Add .gitignore

### PR DESCRIPTION
**Pull request by pilino1234**
Taking recommendations for other stuff that should be added

- [x] Build files when doing in-tree builds

edit: nvm

### CODE REVIEW

**Comment by pilino1234** [comment]
Squashed, good to go 😄

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+# General os files
+*~
+*.bak
+*.swp
+.directory
+.DS_Store
+.DS_Store?
+._*
+.Spotlight-V100
```
Comment (CurlyMoo) [inline .gitignore]: Why?

Comment (pilino1234) [reply .gitignore]: IIRC, it's some index file that OSX uses

Comment (CurlyMoo) [reply .gitignore]: Wierd stuff. And if so, why isn't there a pilight OSX build?

Comment (pilino1234) [reply .gitignore]: Dunno, nobody has tested it yet I suppose. I guess I could remove it for now until people start compiling pilight on mac's 😆

Comment (CurlyMoo) [reply .gitignore]: Yes

Comment (pilino1234) [reply .gitignore]: Done

**Pull request merged**